### PR TITLE
some more es6 changes

### PR DESCRIPTION
**Pull request by stbaer**
- const instead of let if possible
- arrow functions
- string templates
- argument defaults

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
Looks like two tests are failing. Could you look at that? Otherwise, thanks for the additions!

**Comment by stbaer** [comment]
Ya, but they are failing for another pr . In the base branch they are fine...

**Comment by bigtimebuddy** [comment]
Could you rebase with **dev-es6**?

**Comment by englercj** [comment]
Rebase, not merge :P

**Comment by stbaer** [comment]
Oh ^^
I don't really get why/where travis is failing.

**Comment by stbaer** [comment]
Reverted the merge and rebased but travis is still failing.

**Comment by bigtimebuddy** [comment]
@stbaer If you run `npm run test:debug`, you can debug the tests and see where they are failing.

**Comment by stbaer** [comment]
The problem is that they don't fail when I run them locally.

**Comment by bigtimebuddy** [comment]
I cloned your repo and they are failing for me locally when I run `npm run build` or `npm test`

**Comment by stbaer** [comment]
Ok, I didn't run the build first... getting the errors also now. Thank you @bigtimebuddy - will try to fix them.

**Comment by bigtimebuddy** [comment]
Seems to be this section of Text.js

**this works** 

```js
        let properties = Text.fontPropertiesCache[fontStyle];
        if (!properties)
        {
            properties = {};
```

**this fails**
```js
        const properties = Text.fontPropertiesCache[fontStyle] || {};
        if (!properties)
        {
```

**Comment by bigtimebuddy** [comment]
But can you explain why it doesn't work??? 😕

**Comment by stbaer** [comment]
The only difference should be that with the change always at least an empty object will be returned.

File: README.md [hunk 1]
```diff
@@ -14,13 +14,13 @@ across all devices. The Pixi renderer allows everyone to enjoy the power of
 hardware acceleration without prior knowledge of WebGL. Also, it's fast. Really fast.
 
 If you want to keep up to date with the latest pixi.js news then feel free to follow us on twitter
-([@doormat23](https://twitter.com/doormat23), [@bigtimebuddy](https://twitter.com/bigtimebuddy), [@ivanpopelyshev](https://twitter.com/ivanpopelyshev))
+([@doormat23](https://twitter.com/doormat23), and [@rolnaaba](https://twitter.com/rolnaaba))
```
Comment (bigtimebuddy) [inline README.md]: These changes should not be reverted

File: README.md [hunk 2]
```diff
@@ -14,13 +14,13 @@ across all devices. The Pixi renderer allows everyone to enjoy the power of
...
 and we will keep you posted! You can also check back on [our site](http://www.goodboydigital.com/blog)
 as any breakthroughs will be posted up there too!
 
 **Your support helps us make Pixi.js even better. Make your pledge on [Patreon](https://www.patreon.com/user?u=2384552&ty=h&u=2384552) and we'll love you forever!**
 
-[![gallery](http://www.pixijs.com/wp-content/uploads/2013/05/headerPanel_projects-898x342.jpg)](http://www.pixijs.com/gallery/)
+[![projects](http://www.pixijs.com/wp-content/uploads/2013/05/headerPanel_projects-898x342.jpg)](http://www.pixijs.com/projects/)
```
Comment (bigtimebuddy) [inline README.md]: These changes should not be reverted

File: LICENSE [hunk 3]
```diff
@@ -1,6 +1,6 @@
-The Pixi License
+The MIT License
 
-Copyright (c) 2013-2016 Mathew Groves
+Copyright (c) 2013-2015 Mathew Groves
```
Comment (bigtimebuddy) [inline LICENSE]: These changes should not be reverted

**Comment by stbaer** [comment]
@bigtimebuddy - reverted those. Sorry this pr is a bit of a mess ...

**Comment by englercj** [comment]
@bigtimebuddy Can you make sure to squash if/when you merge this?

**Comment by stbaer** [comment]
Yup, done

File: src/core/graphics/utils/bezierCurveTo.js [hunk 4]
```diff
@@ -13,25 +13,21 @@
  * @param cpY2 {number} Second Control point y
  * @param toX {number} Destination point x
  * @param toY {number} Destination point y
- * @param [path=number[]] Path array to push points into
- * @return {PIXI.Graphics}
+ * @param [path=[]] {number[]} Path array to push points into
+ * @return {number[]}
  */
-const bezierCurveTo = function (fromX, fromY, cpX, cpY, cpX2, cpY2, toX, toY, path)
+const bezierCurveTo = function (fromX, fromY, cpX, cpY, cpX2, cpY2, toX, toY, path=[]) // jshint ignore:line
```
Comment (bigtimebuddy) [inline src/core/graphics/utils/bezierCurveTo.js]: Remove `jshint ignore:line`, replaced jshint with eslint

File: src/core/graphics/webgl/utils/buildRoundedRectangle.js [hunk 5]
```diff
@@ -90,28 +89,28 @@ let buildRoundedRectangle = function (graphicsData, webGLData)
  * @param cpY {number} Control point y
  * @param toX {number} Destination point x
  * @param toY {number} Destination point y
- * @param [out] {number[]} The output array to add points into. If not passed, a new array is created.
+ * @param [out=[]] {number[]} The output array to add points into. If not passed, a new array is created.
  * @return {number[]} an array of points
  */
-let quadraticBezierCurve = function (fromX, fromY, cpX, cpY, toX, toY, out)
+let quadraticBezierCurve = function (fromX, fromY, cpX, cpY, toX, toY, out=[])// jshint ignore:line
```
Comment (bigtimebuddy) [inline src/core/graphics/webgl/utils/buildRoundedRectangle.js]: Remove `// jshint ignore:line`

**Comment by bigtimebuddy** [comment]
Small formatting issue, but could you pad the fat arrow with spaces on either side? `(data)=>{}` becomes `(data) => {}`?

**Comment by themoonrat** [comment]
ESLint has a "fix" mode which can automatically sort out spacing / quote formatting issues

**Comment by bigtimebuddy** [comment]
Great work @stbaer, appreciate all the help and back and forth.  This will really help Pixi's conversion to ES6.

**Pull request merged**